src/Frame/Frame.cc

```diff
@@ -1245,39 +1226,25 @@ bool Frame::FillSpatialProfileData(PointXy point, std::vector<CARTA::SetSpatialR
                             have_profile = true;
                         }
                     } else { // Use image cache to return full resolution data or prepare data for decimation
+                        get_spatial_profile_from_cache(CurrentStokes());
+                    }
+                } else {
+                    // Required stokes is not the current stokes or the stokes needs to be computed
+                    if (ImageCacheAvailable(CurrentZ(), stokes)) {
+                        get_spatial_profile_from_cache(stokes);
```

Comment:
`ImageCacheAvailable` is called a second time here inside `get_spatial_profile_from_cache`. But this is an example of logic which should be refactored.